### ISSUE TITLE
Relation mode + `upsert()`: `Internal error: Attempted to serialize empty result.`

### ISSUE THREAD

**Issue by GuoNH** [prisma/prisma#22562]
### Bug description

When the relationMode is configured as Prisma, if there are multiple model relations within a Prisma model bound to the same id field and an upsert method is called to perform an update operation, an error occurs.

log:
```
debug: BEGIN {"params":"[]","source":"PrismaClient","target":"quaint::connector::metrics","timestamp":"2024-01-04T10:07:40.306Z"}
debug: SELECT "graphql"."gg2"."id" FROM "graphql"."gg2" WHERE ("graphql"."gg2"."id" = $1 AND 1=1) OFFSET $2 {"duration":"1 ms","params":"[\"clqyxhu5y0000qpnyssdcmes3\",0]","source":"PrismaClient","target":"quaint::connector::metrics","timestamp":"2024-01-04T10:07:40.308Z"}
debug: SELECT "graphql"."gg6_ttt_aaa"."id", "graphql"."gg6_ttt_aaa"."t2" FROM "graphql"."gg6_ttt_aaa" WHERE (1=1 AND "graphql"."gg6_ttt_aaa"."t2" IN ($1)) OFFSET $2 {"duration":"1 ms","params":"[\"clqyxhu5y0000qpnyssdcmes3\",0]","source":"PrismaClient","target":"quaint::connector::metrics","timestamp":"2024-01-04T10:07:40.310Z"}
debug: SELECT "graphql"."gg3"."id", "graphql"."gg3"."relation" FROM "graphql"."gg3" WHERE (1=1 AND "graphql"."gg3"."relation" IN ($1)) OFFSET $2 {"duration":"1 ms","params":"[\"clqyxhu5y0000qpnyssdcmes3\",0]","source":"PrismaClient","target":"quaint::connector::metrics","timestamp":"2024-01-04T10:07:40.312Z"}
debug: SELECT "graphql"."gg2"."id" FROM "graphql"."gg2" WHERE ("graphql"."gg2"."id" = $1 AND ("graphql"."gg2"."id" = $2 AND 1=1)) LIMIT $3 OFFSET $4 {"params":"[\"clqyxhu5y0000qpnyssdcmes3\",\"clqyxhu5y0000qpnyssdcmes3\",1,0]","source":"PrismaClient","target":"quaint::connector::metrics","timestamp":"2024-01-04T10:07:40.312Z"}
thread 'tokio-runtime-worker' panicked at query-engine\core\src\response_ir\ir_serializer.rs:57:40debug: SELECT "graphql"."gg2"."id" FROM "graphql"."gg2" WHERE "graphql"."gg2"."id" = $1 LIMIT $2 OFFSET $3 {"params":"[\"clqyxhu5y0000qpnyssdcmes3\",1,0]","source":"PrismaClient","target":"quaint::connector::metrics","timestamp":"2024-01-04T10:07:40.313Z"}
:
Internal error: Attempted to serialize empty result.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: 
Invalid `prisma.gg2.upsert()` invocation in
C:\Users\guonaihao\Documents\code\graphql-boilerplate\src\graphql\gg2\mutations\upsertOne.ts:11:23

   8   update: nonNull('gg2UpdateInput'),
   9 },
  10 resolve(_parent, args, { prisma, select }) {
→ 11   return prisma.gg2.upsert(
Internal error: Attempted to serialize empty result.

This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.

https://github.com/prisma/prisma/issues/new?body=Hi+Prisma+Team%21+My+Prisma+Client+just+crashed.+This+is+the+report%3A%0A%23%23+Versions%0A%0A%7C+Name++++++++++++%7C+Version++++++++++++%7C%0A%7C-----------------%7C--------------------%7C%0A%7C+Node++++++++++++%7C+v16.20.0+++++++++++%7C+%0A%7C+OS++++++++++++++%7C+windows++++++++++++%7C%0A%7C+Prisma+Client+++%7C+5.7.1++++++++++++++%7C%0A%7C+Query+Engine++++%7C+0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5%7C%0A%7C+Database++++++++%7C+postgresql+++++++++%7C%0A%0A%0A%0A%23%23+Logs%0A%60%60%60%0Aprisma%3AtryLoadEnv+Environment+variables+loaded+from+C%3A%5CUsers%5Cguonaihao%5CDocuments%5Ccode%5Cgraphql-boilerplate%5C.env%0Aprisma%3Aclient+checkPlatformCaching%3Apostinstall+%0Aprisma%3Aclient+checkPlatformCaching%3AciName+%0Aprisma%3AtryLoadEnv+Environment+variables+loaded+from+C%3A%5CUsers%5Cguonaihao%5CDocuments%5Ccode%5Cgraphql-boilerplate%5C.env%0Aprisma%3Aclient+dirname+C%3A%5CUsers%5Cguonaihao%5CDocuments%5Ccode%5Cgraphql-boilerplate%5Cnode_modules%5C.pnpm%5C%40prisma%2Bclient%405.7.1_prisma%405.7.1%5Cnode_modules%5C.prisma%5Cclient%0Aprisma%3Aclient+relativePath+..%2F..%2F..%2F..%2F..%2F..%2Fprisma%0Aprisma%3Aclient+cwd+C%3A%5CUsers%5Cguonaihao%5CDocuments%5Ccode%5Cgraphql-boilerplate%5Cprisma%0Aprisma%3Aclient+clientVersion+5.7.1%0Aprisma%3Aclient%3AlibraryEngine+internalSetup%0Aprisma%3Aclient%3Aengines%3AresolveEnginePath+enginePath+C%3A%5CUsers%5Cguonaihao%5CDocuments%5Ccode%5Cgraphql-boilerplate%5Cnode_modules%5C.pnpm%5C%40prisma%2Bclient%405.7.1_prisma%405.7.1%5Cnode_modules%5C.prisma%5Cclient%5Cquery_engine-windows.dll.node%0Aprisma%3Aclient%3AlibraryEngine+sending+request%2C+this.libraryStarted%3A+false%0Aprisma%3Aclient%3AlibraryEngine+library+starting%0Aprisma%3Aclient%3AlibraryEngine+library+started%0A%60%60%60%0A%0A%23%23+Client+Snippet%0A%60%60%60ts%0A%2F%2F+PLEASE+FILL+YOUR+CODE+SNIPPET+HERE%0A%60%60%60%0A%0A%23%23+Schema%0A%60%60%60prisma%0A%2F%2F+PLEASE+ADD+YOUR+SCHEMA+HERE+IF+POSSIBLE%0A%60%60%60%0A%0A%23%23+Prisma+Engine+Query%0A%60%60%60%0A%7B%22X%22%3Atrue%7D%7D%7D%0A%60%60%60%0A&title=Internal+error%3A+Attempted+to+serialize+empty+result.&template=bug_report.yml

If you want the Prisma team to look into it, please open the link above 🙏
To increase the chance of success, please post your schema and a snippet of
how you used Prisma Client in the issue.
 {"source":"PrismaClient","target":"gg2.upsert","timestamp":"2024-01-04T10:07:40.324Z"}
```

### How to reproduce

1. Create a Prisma model that establishes a relationship between two models using the id field, with the relationMode configured as `prisma`.
```
datasource db {
  provider = "postgresql"
  url      = "xxx"
  relationMode = "prisma"
}
model gg3 {
  ///
  id       String  @id @default(uuid()) @db.VarChar(255)
  dd       String? @db.VarChar(255)
  gg333    gg2     @relation("gg3_gg333", fields: [relation], references: [id], onDelete: Restrict, onUpdate: Restrict)
  relation String  @unique @db.VarChar(255)
}

model gg2 {
  ggg333        gg3?         @relation("gg3_gg333", map: "23333")
  aa            gg6_ttt_aaa? @relation("gg6_tt_aaa_gg")
  id            String       @id @default(cuid())
  di2           String?
  name          String?
  gg6_ttt_aaaId String?
}

model gg6_ttt_aaa {
  id String  @id @default(cuid())
  t1 String?
  gg gg2?    @relation("gg6_tt_aaa_gg", fields: [t2], references: [id], onDelete: Restrict, onUpdate: Restrict)
  t2 String? @unique
  t3 String?
}

```

2. Call the `gg2.upsert` method, passing in the existing `id` value as the where condition.

### Expected behavior

Return the normal data operation result.

### Prisma information

```prisma
datasource db {
  provider = "postgresql"
  url      = "xxx"
  relationMode = "prisma"
}
model gg3 {
  ///
  id       String  @id @default(uuid()) @db.VarChar(255)
  dd       String? @db.VarChar(255)
  gg333    gg2     @relation("gg3_gg333", fields: [relation], references: [id], onDelete: Restrict, onUpdate: Restrict)
  relation String  @unique @db.VarChar(255)
}

model gg2 {
  ggg333        gg3?         @relation("gg3_gg333", map: "23333")
  aa            gg6_ttt_aaa? @relation("gg6_tt_aaa_gg")
  id            String       @id @default(cuid())
  di2           String?
  name          String?
  gg6_ttt_aaaId String?
}

model gg6_ttt_aaa {
  id String  @id @default(cuid())
  t1 String?
  gg gg2?    @relation("gg6_tt_aaa_gg", fields: [t2], references: [id], onDelete: Restrict, onUpdate: Restrict)
  t2 String? @unique
  t3 String?
}
```

```ts
import { mutationField, nonNull } from 'nexus'

export const Gg2UpsertOneMutation = mutationField('upsertOneGg2', {
  type: nonNull('gg2'),
  args: {
    where: nonNull('gg2WhereUniqueInput'),
    create: nonNull('gg2CreateInput'),
    update: nonNull('gg2UpdateInput'),
  },
  resolve(_parent, args, { prisma, select }) {
    return prisma.gg2.upsert({
      ...args,
      ...select,
    })
  },
})
```

### Environment & setup

- OS: Windows
- Database: PostgreSQL

### Prisma Version

```
prisma                  : 5.7.1
@prisma/client          : 5.7.1
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v16.20.0
Query Engine (Node-API) : libquery-engine 0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5 (at node_modules\.pnpm\@prisma+engines@5.7.1\node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5 (at node_modules\.pnpm\@prisma+engines@5.7.1\node_modules\@prisma\engines\schema-engine-windows.exe)  
Schema Wasm             : @prisma/prisma-schema-wasm 5.7.1-1.0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5
Default Engines Hash    : 0ca5ccbcfa6bdc81c003cf549abe4269f59c41e5
Studio                  : 0.495.0
```

**Comment by laplab** [comment]
Thank you for reporting the issue, @GuoNH! We are currently testing the fix for this problem. Would it be possible for you to check your code against this version of Prisma?
```
npm i prisma@5.13.0-dev.30
```
Bear in mind that this is a development version and should not be used in production. This is only to check if the issue is fixed.

**Comment by laplab** [comment]
This issue was fixed in [Prisma 5.13.0](https://github.com/prisma/prisma/releases/tag/5.13.0).
